pandas/core/indexes/period.py

```diff
@@ -178,115 +169,111 @@ def __new__(cls, data=None, ordinal=None, freq=None, start=None, end=None,
+                           tz=tz, dtype=dtype, copy=copy, **fields)
+        return cls._simple_new(data, name=name)
 
-        # coerce freq to freq object, otherwise it can be coerced elementwise
-        # which is slow
-        if freq:
-            freq = Period._maybe_convert_freq(freq)
+    @property
+    def _ndarray_values(self):
+        return self.values.values
 
-        if data is None:
-            if ordinal is not None:
-                data = np.asarray(ordinal, dtype=np.int64)
-            else:
-                data, freq = cls._generate_range(start, end, periods,
-                                                 freq, fields)
-            return cls._from_ordinals(data, name=name, freq=freq)
-
-        if isinstance(data, PeriodIndex):
-            if freq is None or freq == data.freq:  # no freq change
-                freq = data.freq
-                data = data._ndarray_values
+    # ------------------------------------------------------------------------
+    # Mixin Candidates
+    # err. maybe not...
+
+    @property
+    def _box_func(self):
+        def func(x):
+            if isinstance(x, Period) or x is tslib.NaT:
+                return x
             else:
-                base1, _ = _gfc(data.freq)
-                base2, _ = _gfc(freq)
-                data = period.period_asfreq_arr(data._ndarray_values,
-                                                base1, base2, 1)
-            return cls._simple_new(data, name=name, freq=freq)
-
-        # not array / index
-        if not isinstance(data, (np.ndarray, PeriodIndex,
-                                 DatetimeIndex, Int64Index)):
-            if is_scalar(data) or isinstance(data, Period):
-                cls._scalar_data_error(data)
-
-            # other iterable of some kind
-            if not isinstance(data, (list, tuple)):
-                data = list(data)
-
-            data = np.asarray(data)
-
-        # datetime other than period
-        if is_datetime64_dtype(data.dtype):
-            data = dt64arr_to_periodarr(data, freq, tz)
-            return cls._from_ordinals(data, name=name, freq=freq)
-
-        # check not floats
-        if infer_dtype(data) == 'floating' and len(data) > 0:
-            raise TypeError("PeriodIndex does not allow "
-                            "floating point in construction")
-
-        # anything else, likely an array of strings or periods
-        data = ensure_object(data)
-        freq = freq or period.extract_freq(data)
-        data = period.extract_ordinals(data, freq)
-        return cls._from_ordinals(data, name=name, freq=freq)
+                return Period._from_ordinal(ordinal=x, freq=self.freq)
+        return func
+
+    # ------------------------------------------------------------------------
+    # Straight Dispatch
+
+    @property
+    def freq(self):
+        """Return the frequency object if it is set, otherwise None"""
+        return self.values.freq
```

Comment:
Should we standardize on using `_values` for the underlying EA for now, so some classes (like DatetimeIndex) can still return ndarray for `.values`, but so internally we can use a certain attribute consistently?